### PR TITLE
issue #65 - adding extension dependency for k8s extension

### PR DESCRIPTION
**Pull request by bfitzpat**
Signed-off-by: Brian Fitzpatrick <bfitzpat@redhat.com>

### CODE REVIEW

**Comment by bfitzpat** [comment]
Not sure why travis is failing - looks like a system error on their side though.

**Comment by lhein** [comment]
I tried to run the tests locally but it seems I have no luck either.

```
> vscode-camelk@0.0.6 test /home/lheinema/git/vscode-camelk
> npm run compile && node ./node_modules/vscode/bin/test


> vscode-camelk@0.0.6 compile /home/lheinema/git/vscode-camelk
> tsc -p ./

### VS Code Extension Test Run ###

Current working directory: /home/lheinema/git/vscode-camelk
Found .vscode-test/vscode-1.37.0. Skipping download.
Running extension tests: /home/lheinema/git/vscode-camelk/.vscode-test/vscode-1.37.0/VSCode-linux-x64/code /home/lheinema/git/vscode-camelk/out/test --extensionDevelopmentPath=/home/lheinema/git/vscode-camelk --extensionTestsPath=/home/lheinema/git/vscode-camelk/out/test --locale=en
[main 2019-08-12T07:23:53.783Z] update#setState idle

[before-after-hook]: "Hook()" repurposing warning, use "Hook.Collection()". Read more: https://git.io/upgrade-before-after-hook-to-1.4

(node:20832) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.

Looking for parseable documents...

enabled [ { provider: 'vsls' } ]

[CoenraadS.bracket-pair-colorizer] Accessing a window scoped configuration for a resource is not expected. To associate 'bracketPairColorizer.excludedLanguages' to a resource, define its scope to 'resource' in configuration contributions in 'package.json'.

Found no documents

TypeError: CamelKNodeProvider_1.CamelKNodeProvider is not a constructor

Tests exited with code: 1
npm ERR! Test failed.  See above for more details.

```

Some errors belong to installed extensions. The message ```TypeError: CamelKNodeProvider_1.CamelKNodeProvider is not a constructor``` seems more relevant. Do I miss something to run this as test?

**Comment by bfitzpat** [comment]
Updated typescript version and other versions to try and work around the build issue.

**Comment by bfitzpat** [comment]
So... I am unable to run tests inside VS Code, but can run at a command prompt with npm test now without the weird failure.

**Comment by lhein** [comment]
I see the same error in both console and inside VSCode.

**Comment by bfitzpat** [comment]
After rebasing, we are now past that weird constructor issue and just have one final hurdle. It seems that activating the extension isn't enough to install any extension dependencies.
```
rejected promise not handled within 1 second: Error: Unknown dependency 'ms-kubernetes-tools.vscode-kubernetes-tools'

stack trace: Error: Unknown dependency 'ms-kubernetes-tools.vscode-kubernetes-tools'

    at p._handleActivateRequest (/home/travis/build/camel-tooling/vscode-camelk/.vscode-test/vscode-1.37.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:496:149)

    at p._activateExtensions (/home/travis/build/camel-tooling/vscode-camelk/.vscode-test/vscode-1.37.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:496:607)

    at p.activateByEvent (/home/travis/build/camel-tooling/vscode-camelk/.vscode-test/vscode-1.37.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:494:635)

    at I._activateByEvent (/home/travis/build/camel-tooling/vscode-camelk/.vscode-test/vscode-1.37.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:760:680)

    at I._handleEagerExtensions (/home/travis/build/camel-tooling/vscode-camelk/.vscode-test/vscode-1.37.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:764:710)

    at define._startExtensionHost._readyToStartExtensionHost.wait.then.then (/home/travis/build/camel-tooling/vscode-camelk/.vscode-test/vscode-1.37.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:768:386)

[main 2019-08-19T18:00:11.208Z] update#setState checking for updates

[main 2019-08-19T18:00:11.267Z] update#setState idle

      1) vscode-camelk extension should activate
```

**Comment by bfitzpat** [comment]
Doing some investigating, there does not seem to be a way to test for activation when there's a dependent extension. Looking at https://github.com/Azure/vscode-kubernetes-tools/blob/master/test/extension.test.ts, for example - all those tests have been disabled for some reason and there is no activation test. 

I have opened a question on Stack Exchange to see if anything comes up - https://stackoverflow.com/questions/57562000/is-there-a-way-to-activate-a-dependent-extension-to-test-for-full-extension-acti

But for the meantime, we may be able to simply disable this test until we have a better way to handle it. All the other tests run successfully.

**Comment by bfitzpat** [comment]
After revising things a bit and rebasing, we seem to be in better shape with this. All tests pass. Open to suggestions as always

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,1084 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (apupier) [inline yarn.lock]: I think the yarn file should not be present.

**Comment by bfitzpat** [comment]
> as spotted by Brian, the drop down list of start integration command are no more working. they are vanishing immediately.

This was fixed by two other PRs that were introduced. Thanks @apupier ! :)

**Comment by bfitzpat** [comment]
@apupier Though you approved this, I'm going to update it to add a comment to the changelog. It should be documented I think

**Pull request merged**